File: backend-java/src/main/java/com/groceriesapp/service/HouseholdService.java
package com.groceriesapp.service;

import com.groceriesapp.model.Household;
import com.groceriesapp.repository.HouseholdRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class HouseholdService {
    
    private final HouseholdRepository householdRepository;
    
    public List<Household> getAllHouseholds() {
        return householdRepository.findAll();
    }
    
    public Household getHouseholdById(Long id) {
        return householdRepository.findById(id).orElse(null);
    }
}

File: backend/src/main/kotlin/com/groceriesapp/service/HouseholdService.kt
package com.groceriesapp.service

import com.groceriesapp.model.Household
import com.groceriesapp.repository.HouseholdRepository
import org.springframework.stereotype.Service

@Service
class HouseholdService(
    private val householdRepository: HouseholdRepository
) {
    fun findAll(): List<Household> = householdRepository.findAll()

    fun findById(id: Long): Household? = householdRepository.findById(id).orElse(null)
}
